monitoring/boot_config.go

```diff
@@ -0,0 +1,186 @@
+	reBootConfigComment = regexp.MustCompile(`#.*`)
+)
+
+func parseBootConfig() (map[string]string, error) {
+	var uname syscall.Utsname
+	err := syscall.Uname(&uname)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	file, err := os.Open(fmt.Sprintf("/boot/config-%s", int8string(uname.Release[:])))
+	if err != nil {
+		return nil, trace.ConvertSystemError(err)
+	}
+	defer file.Close()
+
+	params := map[string]string{}
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		if reBootConfigComment.Match(scanner.Bytes()) || scanner.Text() == "" {
+			continue
+		}
+
+		parsed := reBootConfigParam.FindStringSubmatch(scanner.Text())
+		if len(parsed) != 3 {
+			continue
+		}
+
+		params[parsed[1]] = parsed[2]
+	}
+
+	err = scanner.Err()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return params, nil
+}
+
+// BootConfigParam defines parameter name (without CONFIG_ prefix) to check
+type BootConfigParam struct {
+	// Param is boot config parameter to check
+	Name string
+}
+
+// BootConfigParamChecker checks whether parameters provided are specified in linux boot configuration file
+type BootConfigParamChecker struct {
+	// Params is array of parameters to check for
+	Params []BootConfigParam
+}
+
+const (
+	bootConfigParamID = "boot-config"
+)
+
+// Name returns name of the checker
+func (c *BootConfigParamChecker) Name() string {
+	return bootConfigParamID
+}
+
+// Check parses boot config files and validates whether parameters provided are set
+func (c *BootConfigParamChecker) Check(ctx context.Context, reporter health.Reporter) {
+	cfg, err := parseBootConfig()
+	if trace.IsNotFound(err) {
+		reporter.Add(NewProbeFromErr(bootConfigParamID,
+			"boot config unavailable, checks skipped", err))
+		return
+	}
+	if err != nil {
+		reporter.Add(NewProbeFromErr(bootConfigParamID, "failed to parse boot config file", trace.Wrap(err)))
+		return
+	}
+
+	failed := false
+	for _, param := range c.Params {
+		if _, ok := cfg[param.Name]; ok {
+			continue
+		}
+
+		reporter.Add(&pb.Probe{
+			Checker: bootConfigParamID,
+			Detail:  fmt.Sprintf("required kernel boot config parameter %s missing", param.Name),
+			Status:  pb.Probe_Failed,
+		})
+		failed = true
+	}
+
+	if failed {
+		return
+	}
+	reporter.Add(&pb.Probe{
+		Checker: bootConfigParamID,
+		Status:  pb.Probe_Running,
+	})
+}
+
+// DefaultBootConfigParams returns standard kernel configs required for running kubernetes
+func DefaultBootConfigParams() *BootConfigParamChecker {
+	return &BootConfigParamChecker{
+		[]BootConfigParam{
+			BootConfigParam{"CONFIG_NET_NS"},
+			BootConfigParam{"CONFIG_PID_NS"},
+			BootConfigParam{"CONFIG_IPC_NS"},
+			BootConfigParam{"CONFIG_UTS_NS"},
+			BootConfigParam{"CONFIG_CGROUPS"},
+			BootConfigParam{"CONFIG_CGROUP_CPUACCT"},
+			BootConfigParam{"CONFIG_CGROUP_DEVICE"},
+			BootConfigParam{"CONFIG_CGROUP_FREEZER"},
+			BootConfigParam{"CONFIG_CGROUP_SCHED"},
+			BootConfigParam{"CONFIG_CPUSETS"},
+			BootConfigParam{"CONFIG_MEMCG"},
```

Comment:
I think this is not enough - although the kernel is configured with memory cgroup support, it can still be turned off in the boot config in `/boot/grub/grub.cfg` (see https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/resource_management_guide/sec-memory).
I think the only confident way is checking cgroups is `/proc/cgroups` or cgroup mounts in `/proc/mounts`

Reply:
yes, that's separate checker though